doc/interfaces/socketcand.rst

```diff
@@ -0,0 +1,113 @@
+.. _socketcand_doc:
+
+socketcand Interface
+====================
+`Socketcand <https://github.com/linux-can/socketcand>`__ is part of the 
+`Linux-CAN <https://github.com/linux-can>`__ project, providing a 
+Network-to-CAN bridge as Linux damon. It implements a specific
+`TCP/IP based communication protocol <https://github.com/linux-can/socketcand/blob/master/doc/protocol.md>`__
+to transfer CAN frames and control commands.
+
+The main advantage compared to UDP-based protocols (e.g. virtual interface)
+is, that TCP guarantees delivery and that the message order is kept.
+
+Here is a small example dumping all can messages received by a socketcand 
+daemon running on a remote Raspberry Pi:
+
+.. code-block:: python
+
+    import can
+
+    bus = can.interface.Bus(bustype='socketcand', host="10.0.16.15", port=29536, channel="can0")
+
+    # loop until Ctrl-C
+    try:
+      while True:
+        msg = bus.recv()
+        print (msg)
+    except KeyboardInterrupt:
+      pass
+
+The output may look like this:
+::
+    Timestamp: 1637791111.209224    ID: 000006fd    X Rx                DLC:  8    c4 10 e3 2d 96 ff 25 6b
+    Timestamp: 1637791111.233951    ID: 000001ad    X Rx                DLC:  4    4d 47 c7 64
+    Timestamp: 1637791111.409415    ID: 000005f7    X Rx                DLC:  8    86 de e6 0f 42 55 5d 39
+    Timestamp: 1637791111.434377    ID: 00000665    X Rx                DLC:  8    97 96 51 0f 23 25 fc 28
+    Timestamp: 1637791111.609763    ID: 0000031d    X Rx                DLC:  8    16 27 d8 3d fe d8 31 24
+    Timestamp: 1637791111.634630    ID: 00000587    X Rx                DLC:  8    4e 06 85 23 6f 81 2b 65
+
+The following section will show how to get the stuff installed on a Raspberry Pi with a MCP2515-based
+CAN interface, e.g. available from Waveshare: https://www.waveshare.com/rs485-can-hat.htm .
+
+Install CAN Interface for a MCP2515 based interface on a Raspberry Pi
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Add the following lines to ``/boot/config.txt``. Please take care on the frequencsy of the crystal on your MCP2515 board:
+::
+    dtparam=spi=on
+    dtoverlay=mcp2515-can0,oscillator=12000000,interrupt=25,spimaxfrequency=1000000
+
+Reboot after ``/boot/config.txt`` has been modified.
+
+
+Enable socketcan for can0
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Create config file for systemd-networkd to start the socketcan interface automatically:
+
+.. code-block:: bash
+
+    cat >/etc/systemd/network/80-can.network <<'EOT'
+    [Match]
+    Name=can0
+    [CAN]
+    BitRate=250K
+    RestartSec=100ms
+    EOT
+
+Enable ``systemd-networkd`` on reboot and start it immediately (if it was not already startet):
+
+.. code-block:: bash
+
+    sudo systemctl enable systemd-networkd
+    sudo systemctl start systemd-networkd
+
```

Comment:
This is in principle a nice tutorial, but it does not really explain the interface in particular. (Do you agree?) We could also just reference the socketcan page here, since any set up socketcan interface would work, right?

Reply:
Yes, the tutorial provides much more information than really necessary. The idea was to give a full setup tutorial, what is currently hard to find, and what is also not provided by sockencand. But I can remove this section if you want.

Reply:
It is a useful and well-written tutorial indeed! So first of all, thank you for contributing it.

It's fine if we add a few sentences on it working with any socketcan connection, not just the one you mentioned. It should be clear that it is only an example.

Reply:
Ok, I've split it off a bit as a separate section "Socketcand Quickstart", as it is done in the socketcan interface doc.